OnPageCode_timeout.js

```diff
@@ -1,288 +1,457 @@
-      "headers": {
-        "content-type": "application/json"
-      },
-      "method": "POST",
-      "body": JSON.stringify(jsonData)
-    })
-      .then(res => { })
-      .then(response => { })
-      .catch(error => { });
-  },
-
-  getParameterByName: function (name, url) {
-    name = name.replace(/[\[\]]/g, '\\$&');
-    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
-      results = regex.exec(url);
-    if (!results)
-      return null;
-    if (!results[2])
-      return '';
-    return decodeURIComponent(results[2].replace(/\+/g, ' '));
-  },
-
-  prepareNetworkLatencyData: function (perfResource, bidderRequest, latency) {
-    let output = {
-      domain: PM_Network_POC.domain,
-      publisherId: PM_Network_POC.publisherId,
-      browser: PM_Network_POC.browserName,
-      timestamp: Date.now(),
-      bidder: {
-        name: bidderRequest.bidderCode,
-        request: {
-          method: bidderRequest?.nwMonitor?.reqMethod,
-          isOverride: bidderRequest?.nwMonitor?.reqOverride,
-          endPoint: bidderRequest?.nwMonitor?.reqEndPoint
-        }
-      },
-      nw: {
-        evaluated: {},
+
+    'bidCallTracker': [],
+
+    prepareNetworkLatencyData: function (perfResource, bidderRequest, sspConf, latency, processedBidsCount) {
+      let currentBidCall = {};
+      let output = {
+        domain: PM_Network_POC.domain,
+        publisherId: PM_Network_POC.publisherId,
+        browser: PM_Network_POC.browserName,
+        timestamp: Date.now(),
+        platform: PM_Network_POC.platform,
+        adUnitCount: PM_Network_POC.adUnitCount,
+        atLeastOneBidResUsedInAuction: sspConf.atLeastOneBidResUsedInAuction,
+        bidder: {
+          name: sspConf.bidderCode,
+          requests: PM_Network_POC.bidCallTracker,
+        },
+        serverLatency: latency || {},
+        t: PM_Network_POC.timeoutCorrelators[bidderRequest?.nwMonitor?.correlator] ? 1 : 0,
+        db: perfResource?.name?.length ? 0 : 1
+      };
+
+      if (PM_Network_POC.networkType) output['networkType'] = PM_Network_POC.networkType;
+
+      currentBidCall.isOverride = bidderRequest?.nwMonitor?.reqOverride;
+      currentBidCall.endPoint = bidderRequest?.nwMonitor?.reqEndPoint;
+
+      currentBidCall.order = sspConf.bids[processedBidsCount - 1].order;
+      currentBidCall.method = sspConf.bids[processedBidsCount - 1].reqMethod;
+      currentBidCall.nw = {
+        evaluated: {
+          // percieved latency
+          per_lt: Date.now() - PM_Network_POC.cache.auction.timestamp
+        },
         raw: {}
-      },
-      serverLatency: latency || {},
-      requestUrlPayloadLength: bidderRequest?.nwMonitor?.requestUrlPayloadLength,
-      t: PM_Network_POC.timeoutCorrelators[bidderRequest?.nwMonitor?.correlator] ? 1 : 0,
-      db: perfResource?.name?.length ? 0 : 1
-    };
+      };
+      currentBidCall.requestUrlPayloadLength = bidderRequest?.nwMonitor?.requestUrlPayloadLength;
```

Comment:
Line 309 is specific to PubMatic and captures the payload length (which includes the pubmatic url and payload).  

Line 310 is taken from pbjs and only captures the length of the payload (not additionally including the length of the url too, just the payload).  This is captured for all bid requests from pbjs.

Question is, how should we capture this payload length data?  should we capture this within 2 separate fields?  or under one field, maybe something like the following?

`currentBidCall.reqPayloadLength =
    bidderRequest?.nwMonitor?.requestUrlPayloadLength
        ? bidderRequest?.nwMonitor?.requestUrlPayloadLength
        : sspConf.bids[processedBidsCount - 1].reqPayloadCharCount;`